ui/apps/platform/src/Containers/SystemConfig/SystemConfigForm/SystemConfigForm.tsx

```diff
@@ -17,31 +19,75 @@ import {
     CardActions,
     Switch,
 } from '@patternfly/react-core';
+import { useFormik } from 'formik';
 
 import ColorPicker from 'Components/ColorPicker';
 import { getProductBranding } from 'constants/productBranding';
-import { PrivateConfig, PublicConfig } from 'types/config.proto';
+import { PrivateConfig, PublicConfig, SystemConfig } from 'types/config.proto';
 import { TelemetryConfig } from 'types/telemetry.proto';
 import { ConfigTelemetryDetailContent } from '../ConfigTelemetryDetailWidget';
 import FormSelect from './FormSelect';
 
-export type SystemConfigFormProps = {
-    values: {
-        privateConfig: PrivateConfig;
-        publicConfig: PublicConfig;
-        telemetryConfig: TelemetryConfig;
+function getCompletePublicConfig(systemConfig: SystemConfig): PublicConfig {
+    return {
+        header: {
+            color: systemConfig?.publicConfig?.header?.color || '#000000',
+            backgroundColor: systemConfig?.publicConfig?.header?.backgroundColor || '#FFFFFF',
+            text: systemConfig?.publicConfig?.header?.text || '',
+            enabled: systemConfig?.publicConfig?.header?.enabled || false,
+            size: systemConfig?.publicConfig?.header?.size || 'UNSET',
+        },
+        footer: {
+            color: systemConfig?.publicConfig?.footer?.color || '#000000',
+            backgroundColor: systemConfig?.publicConfig?.footer?.backgroundColor || '#FFFFFF',
+            text: systemConfig?.publicConfig?.footer?.text || '',
+            enabled: systemConfig?.publicConfig?.footer?.enabled || false,
+            size: systemConfig?.publicConfig?.footer?.size || 'UNSET',
+        },
+        loginNotice: {
+            text: systemConfig?.publicConfig?.loginNotice?.text || '',
+            enabled: systemConfig?.publicConfig?.loginNotice?.enabled || false,
+        },
```

Comment:
Super suggestion!

Reply:
Aha, I will do as follow up, because possible `null` values need careful attention.

> Source properties that resolve to undefined are skipped if a destination value exists.